### ISSUE TITLE
[Feature Request] refresh_in 

### ISSUE THREAD

**Issue by henrik-me** [AzureAD/microsoft-authentication-library-for-python#176]
Refresh_In (token response) support. 
This feature allows the service to control when MSAL should attempt to refresh the access token (ahead of its expiration)

[API Review](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FRefreshAtExpirationPercentage%2Foverview.md&version=GBdev&_a=contents)

**Comment by lzw-all-in** [comment]
Hi @henrik-me  I just have the same request. I need to get new valid access token through my refresh token.

**Comment by rayluo** [comment]
@lzw-all-in , you can already get a new valid access token through using `acquire_token_silent()`. Please refer to [the 3-step pattern described in our README](https://github.com/AzureAD/microsoft-authentication-library-for-python#usage), and then you can read our API Reference located at the header line in our repo.

This feature described here is for msal to FURTHER improve our INTERNAL refresh logic, but that is transparent to you as an app developer. Yoiu are NOT blocked by this feature.

**Comment by lzw-all-in** [comment]
@rayluo Thank you. I saw your usage and source code before but what I need is the api I can pass refresh token as a parameter to get the newest access token.

**Comment by rayluo** [comment]
@lzw-all-in , do you intend to "use oauth2 api" but still connect with Microsoft identity platform (i.e. Azure Active Directory)? If so, MSAL is designed to be easier than using generic OAuth2 library. Or if you have any specific need to handle refresh token yourself, I would be curious to hear that.

Either way, that would be a different topic which does not belong to this Feature Request. Feel free to create a separated issue, and/or send email to me. Thanks.

**Comment by lzw-all-in** [comment]
@rayluo  Yes I saw MSAL api doesn't support given refresh token to attain the newest access token. I use the api `obtain_token_by_refresh_token` supported by `oauth2` Client class.
Because for every client who use my app service I can get their access token and refresh token. And every time when I found the access token is failed then I will use this user's refresh token to get a new one.
```python
result = Azureapp.client.obtain_token_by_refresh_token(
            token_item={"refresh_token" : refresh_token},
            scope=app_config.RESOURCE)
```
